`TestModeratedSessionWithMFA` flakiness

## Failure

#### Link(s) to logs

- https://github.com/gravitational/teleport/actions/runs/5312106326/jobs/9616179371#step:6:1268

#### Relevant snippet

It's not a typical flaky test. What fails is the cleanup step.

```
     testing.go:1206: TempDir RemoveAll cleanup: unlinkat /tmp/TestModeratedSessionWithMFA3205863439/002/log/upload/streaming/default/multi/f49e220a-8667-479f-bddf-13a3bae78732/d8b867ea-00f4-4d4b-881b-d631723a3120: directory not empty
--- FAIL: TestModeratedSessionWithMFA (6.48s)
```

--- a/lib/web/apiserver_test.go
+++ b/lib/web/apiserver_test.go
@@ -8964,7 +8964,7 @@ func TestModeratedSession(t *testing.T) {
 	modules.SetTestModules(t, &modules.TestModules{TestBuildType: modules.BuildEnterprise})
 
 	ctx := context.Background()
-	s := newWebSuite(t)
+	s := newWebSuiteWithConfig(t, webSuiteConfig{disableDiskBasedRecording: true})
 
 	peerRole, err := types.NewRole("moderated", types.RoleSpecV6{
 		Allow: types.RoleConditions{
@@ -9042,6 +9042,7 @@ func TestModeratedSessionWithMFA(t *testing.T) {
 	const RPID = "localhost"
 
 	s := newWebSuiteWithConfig(t, webSuiteConfig{
+		disableDiskBasedRecording: true,
 		authPreferenceSpec: &types.AuthPreferenceSpecV2{
 			Type:           constants.Local,
 			ConnectorName:  constants.PasswordlessConnector,

